toolkit/tools/imagegen/diskutils/verity.go

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Utility to create read-only root partitions
+
+package diskutils
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"microsoft.com/pkggen/imagegen/configuration"
+	"microsoft.com/pkggen/internal/logger"
+	"microsoft.com/pkggen/internal/randomization"
+	"microsoft.com/pkggen/internal/shell"
+)
+
+const (
+	mappingVerityPrefix = "verity-"
+	debugMountPoint     = "/mnt/verity_overlay_debug_tmpfs"
+)
+
+// VerityDevice represents a device mapper linear device used for a dm-verity read-only partition.
+// - MappedName is the desired device mapper name
+// - MappedDevice is the full path of the created device mapper device
+// - BackingDevice is the underlying file/device which backs the partition
+// - FecRoots is the number of error correcting roots, 0 to omit error correction
+// - ValidateOnBoot will cause a full, user-mode analysis of the verity disk during boot (good for debugging)
+// - UseRootHashSignature indicates a signature file has been included with the verity disk and should be checked
+// - ErrorBehavior is what dm-verity should do in the event of corruption (ignore, panic, restart)
+// - TmpfsOverlays is a list of tmpfs overlays which will be created after the verity partition is mounted
+// - TmpfsOverlaysDebugMount indicates if the overlays should be made accessible for debugging purposes
+type VerityDevice struct {
+	MappedName              string
+	MappedDevice            string
+	BackingDevice           string
+	FecRoots                int
+	ValidateOnBoot          bool
+	UseRootHashSignature    bool
+	ErrorBehavior           string
+	TmpfsOverlays           []string
+	TmpfsOverlaysDebugMount string
+}
+
+// AddRootVerityFilesToInitramfs adds files needed for a verity root to the initramfs
+// - workingFolder is a temporary folder to extract the initramfs to
+// - initramfsPath is the path to the initramfs
+func (v *VerityDevice) AddRootVerityFilesToInitramfs(workingFolder, initramfsPath string) (err error) {
+	var (
+		verityWorkingDirectory = filepath.Join(workingFolder, v.MappedName)
```

Comment:
Nit, since assignment is happening here, use the `:=` variable initialization syntax.

```golang
verityWorkingDirectory := filepath.Join(workingFolder, v.MappedName)
````